lessons/basics/typespec.md

```diff
@@ -0,0 +1,197 @@
+---
+layout: page
+title: Specifications and types
+category: basics
```

Comment:
I agree. Even if coming from something like Java and seeking this out immediately, its far from anything required to function.

I think things like the `h` helper that access docs/specs/types are wonderful for the basics, but actually making them and leveraging what they offer etc is something I've never seen come as a core competency for anyone who is just starting to learn.